<?xml version="1.0" encoding="UTF-8" standalone="no"?>
<testsuite errors="0" failures="6" name="io.cucumber.core.plugin.JUnitFormatter" skipped="0" tests="11" time="541.656">
<testcase classname="F01_Register | users could register with new accounts" name="guest user could register with valid data successfully" time="398.756">
<failure message="java.util.concurrent.TimeoutException&#10;Build info: version: '4.3.0', revision: 'a4995e2c09*'&#10;System info: host: 'DESKTOP-9J129ML', ip: '192.168.1.12', os.name: 'Windows 10', os.arch: 'amd64', os.version: '10.0', java.version: '1.8.0_281'&#10;Driver info: org.openqa.selenium.chrome.ChromeDriver&#10;Command: [efc2ef7f5dfd53a0f9209c922dfae9a5, get {url=https://demo.nopcommerce.com/}]&#10;Capabilities {acceptInsecureCerts: false, browserName: chrome, browserVersion: 103.0.5060.114, chrome: {chromedriverVersion: 103.0.5060.53 (a1711811edd7..., userDataDir: C:\Users\ENGAMR~1\AppData\L...}, goog:chromeOptions: {debuggerAddress: localhost:58245}, networkConnectionEnabled: false, pageLoadStrategy: normal, platformName: WINDOWS, proxy: Proxy(), se:cdp: ws://localhost:58245/devtoo..., se:cdpVersion: 103.0.5060.114, setWindowRect: true, strictFileInteractability: false, timeouts: {implicit: 0, pageLoad: 300000, script: 30000}, unhandledPromptBehavior: dismiss and notify, webauthn:extension:credBlob: true, webauthn:extension:largeBlob: true, webauthn:virtualAuthenticators: true}&#10;Session ID: efc2ef7f5dfd53a0f9209c922dfae9a5" type="org.openqa.selenium.TimeoutException"><![CDATA[Given user go to register page..............................................skipped
When user select gender type................................................skipped
And user enter first name "automation" and last name "tester"...............skipped
And user enter date of birth................................................skipped
And user enter email "test@example.com" field...............................skipped
And user fills Password fields "P@ssw0rd" "P@ssw0rd"........................skipped
And user clicks on register button..........................................skipped
Then success message is displayed...........................................skipped

StackTrace:
org.openqa.selenium.TimeoutException: java.util.concurrent.TimeoutException
Build info: version: '4.3.0', revision: 'a4995e2c09*'
System info: host: 'DESKTOP-9J129ML', ip: '192.168.1.12', os.name: 'Windows 10', os.arch: 'amd64', os.version: '10.0', java.version: '1.8.0_281'
Driver info: org.openqa.selenium.chrome.ChromeDriver
Command: [efc2ef7f5dfd53a0f9209c922dfae9a5, get {url=https://demo.nopcommerce.com/}]
Capabilities {acceptInsecureCerts: false, browserName: chrome, browserVersion: 103.0.5060.114, chrome: {chromedriverVersion: 103.0.5060.53 (a1711811edd7..., userDataDir: C:\Users\ENGAMR~1\AppData\L...}, goog:chromeOptions: {debuggerAddress: localhost:58245}, networkConnectionEnabled: false, pageLoadStrategy: normal, platformName: WINDOWS, proxy: Proxy(), se:cdp: ws://localhost:58245/devtoo..., se:cdpVersion: 103.0.5060.114, setWindowRect: true, strictFileInteractability: false, timeouts: {implicit: 0, pageLoad: 300000, script: 30000}, unhandledPromptBehavior: dismiss and notify, webauthn:extension:credBlob: true, webauthn:extension:largeBlob: true, webauthn:virtualAuthenticators: true}
Session ID: efc2ef7f5dfd53a0f9209c922dfae9a5
	at org.openqa.selenium.remote.http.netty.NettyHttpHandler.makeCall(NettyHttpHandler.java:65)
	at org.openqa.selenium.remote.http.AddSeleniumUserAgent.lambda$apply$0(AddSeleniumUserAgent.java:42)
	at org.openqa.selenium.remote.http.Filter.lambda$andFinally$1(Filter.java:56)
	at org.openqa.selenium.remote.http.netty.NettyHttpHandler.execute(NettyHttpHandler.java:49)
	at org.openqa.selenium.remote.http.AddSeleniumUserAgent.lambda$apply$0(AddSeleniumUserAgent.java:42)
	at org.openqa.selenium.remote.http.Filter.lambda$andFinally$1(Filter.java:56)
	at org.openqa.selenium.remote.http.netty.NettyClient.execute(NettyClient.java:98)
	at org.openqa.selenium.remote.HttpCommandExecutor.execute(HttpCommandExecutor.java:181)
	at org.openqa.selenium.remote.service.DriverCommandExecutor.invokeExecute(DriverCommandExecutor.java:167)
	at org.openqa.selenium.remote.service.DriverCommandExecutor.execute(DriverCommandExecutor.java:142)
	at org.openqa.selenium.remote.RemoteWebDriver.execute(RemoteWebDriver.java:569)
	at org.openqa.selenium.remote.RemoteWebDriver.get(RemoteWebDriver.java:332)
	at org.openqa.selenium.remote.RemoteWebDriver$RemoteNavigation.to(RemoteWebDriver.java:1056)
	at org.example.stepDefs.Hooks.openBrowser(Hooks.java:20)
Caused by: java.util.concurrent.TimeoutException
	at java.util.concurrent.CompletableFuture.timedGet(CompletableFuture.java:1784)
	at java.util.concurrent.CompletableFuture.get(CompletableFuture.java:1928)
	at org.asynchttpclient.netty.NettyResponseFuture.get(NettyResponseFuture.java:206)
	at org.openqa.selenium.remote.http.netty.NettyHttpHandler.makeCall(NettyHttpHandler.java:59)
	at org.openqa.selenium.remote.http.AddSeleniumUserAgent.lambda$apply$0(AddSeleniumUserAgent.java:42)
	at org.openqa.selenium.remote.http.Filter.lambda$andFinally$1(Filter.java:56)
	at org.openqa.selenium.remote.http.netty.NettyHttpHandler.execute(NettyHttpHandler.java:49)
	at org.openqa.selenium.remote.http.AddSeleniumUserAgent.lambda$apply$0(AddSeleniumUserAgent.java:42)
	at org.openqa.selenium.remote.http.Filter.lambda$andFinally$1(Filter.java:56)
	at org.openqa.selenium.remote.http.netty.NettyClient.execute(NettyClient.java:98)
	at org.openqa.selenium.remote.HttpCommandExecutor.execute(HttpCommandExecutor.java:181)
	at org.openqa.selenium.remote.service.DriverCommandExecutor.invokeExecute(DriverCommandExecutor.java:167)
	at org.openqa.selenium.remote.service.DriverCommandExecutor.execute(DriverCommandExecutor.java:142)
	at org.openqa.selenium.remote.RemoteWebDriver.execute(RemoteWebDriver.java:569)
	at org.openqa.selenium.remote.RemoteWebDriver.get(RemoteWebDriver.java:332)
	at org.openqa.selenium.remote.RemoteWebDriver$RemoteNavigation.to(RemoteWebDriver.java:1056)
	at org.example.stepDefs.Hooks.openBrowser(Hooks.java:20)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at io.cucumber.java.Invoker.doInvoke(Invoker.java:66)
	at io.cucumber.java.Invoker.invokeStatic(Invoker.java:80)
	at io.cucumber.java.AbstractGlueDefinition.invokeMethod(AbstractGlueDefinition.java:45)
	at io.cucumber.java.JavaHookDefinition.execute(JavaHookDefinition.java:64)
	at io.cucumber.core.runner.CoreHookDefinition.execute(CoreHookDefinition.java:46)
	at io.cucumber.core.runner.HookDefinitionMatch.runStep(HookDefinitionMatch.java:21)
	at io.cucumber.core.runner.ExecutionMode$1.execute(ExecutionMode.java:10)
	at io.cucumber.core.runner.TestStep.executeStep(TestStep.java:85)
	at io.cucumber.core.runner.TestStep.run(TestStep.java:57)
	at io.cucumber.core.runner.TestCase.run(TestCase.java:78)
	at io.cucumber.core.runner.Runner.runPickle(Runner.java:75)
	at io.cucumber.testng.TestNGCucumberRunner.lambda$runScenario$1(TestNGCucumberRunner.java:132)
	at io.cucumber.core.runtime.CucumberExecutionContext.lambda$runTestCase$5(CucumberExecutionContext.java:129)
	at io.cucumber.core.runtime.RethrowingThrowableCollector.executeAndThrow(RethrowingThrowableCollector.java:23)
	at io.cucumber.core.runtime.CucumberExecutionContext.runTestCase(CucumberExecutionContext.java:129)
	at io.cucumber.testng.TestNGCucumberRunner.runScenario(TestNGCucumberRunner.java:129)
	at io.cucumber.testng.AbstractTestNGCucumberTests.runScenario(AbstractTestNGCucumberTests.java:35)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.testng.internal.invokers.MethodInvocationHelper.invokeMethod(MethodInvocationHelper.java:135)
	at org.testng.internal.invokers.TestInvoker.invokeMethod(TestInvoker.java:673)
	at org.testng.internal.invokers.TestInvoker.invokeTestMethod(TestInvoker.java:220)
	at org.testng.internal.invokers.MethodRunner.runInSequence(MethodRunner.java:50)
	at org.testng.internal.invokers.TestInvoker$MethodInvocationAgent.invoke(TestInvoker.java:945)
	at org.testng.internal.invokers.TestInvoker.invokeTestMethods(TestInvoker.java:193)
	at org.testng.internal.invokers.TestMethodWorker.invokeTestMethods(TestMethodWorker.java:146)
	at org.testng.internal.invokers.TestMethodWorker.run(TestMethodWorker.java:128)
	at java.util.ArrayList.forEach(ArrayList.java:1259)
	at org.testng.TestRunner.privateRun(TestRunner.java:808)
	at org.testng.TestRunner.run(TestRunner.java:603)
	at org.testng.SuiteRunner.runTest(SuiteRunner.java:429)
	at org.testng.SuiteRunner.runSequentially(SuiteRunner.java:423)
	at org.testng.SuiteRunner.privateRun(SuiteRunner.java:383)
	at org.testng.SuiteRunner.run(SuiteRunner.java:326)
	at org.testng.SuiteRunnerWorker.runSuite(SuiteRunnerWorker.java:52)
	at org.testng.SuiteRunnerWorker.run(SuiteRunnerWorker.java:95)
	at org.testng.TestNG.runSuitesSequentially(TestNG.java:1249)
	at org.testng.TestNG.runSuitesLocally(TestNG.java:1169)
	at org.testng.TestNG.runSuites(TestNG.java:1092)
	at org.testng.TestNG.run(TestNG.java:1060)
	at com.intellij.rt.testng.IDEARemoteTestNG.run(IDEARemoteTestNG.java:66)
	at com.intellij.rt.testng.RemoteTestNGStarter.main(RemoteTestNGStarter.java:109)
]]></failure>
</testcase>
<testcase classname="F02_Login | users could use login functionality to use their accounts" name="user could login with valid email and password" time="22.588">
<failure message="The following asserts failed:&#10;&#9;expected [https://demo.nopcommerce.com/] but found [https://demo.nopcommerce.com/login?returnurl=%2F]" type="java.lang.AssertionError"><![CDATA[Given user go to login page.................................................passed
When user login with valid "test104@example.com"............................passed
And valid "P@ssw0rd"........................................................passed
And user press on login button..............................................passed
Then user login to the system successfully..................................failed

StackTrace:
java.lang.AssertionError: The following asserts failed:
	expected [https://demo.nopcommerce.com/] but found [https://demo.nopcommerce.com/login?returnurl=%2F]
	at org.testng.asserts.SoftAssert.assertAll(SoftAssert.java:46)
	at org.testng.asserts.SoftAssert.assertAll(SoftAssert.java:30)
	at org.example.stepDefs.D02_loginStepDef.userLoginToTheSystemSuccessfully(D02_loginStepDef.java:57)
	at ✽.user login to the system successfully(file:///C:/apache-maven-3.8.6/demonopcommerce/src/main/resources/features/F02_Login.feature:9)
]]></failure>
</testcase>
<testcase classname="F02_Login | users could use login functionality to use their accounts" name="user could login with invalid email and password" time="19.233">
<system-out><![CDATA[Given user go to login page.................................................passed
When user login with invalid "test000@example.com"..........................passed
And invalid "P@ssw0rd"......................................................passed
And user press on login button..............................................passed
Then user could not login to the system.....................................passed
]]></system-out>
</testcase>
<testcase classname="Change Currency Feature" name="user could change currency feature" time="14.779">
<system-out><![CDATA[Given user go to home page..................................................passed
When Select Euro currency from the dropdown list............................passed
Then Euro Symbol (€) is shown on the four products..........................passed
]]></system-out>
</testcase>
<testcase classname="Search Feature" name="user could search using product name" time="10.062">
<failure message="Index: 1, Size: 1" type="java.lang.IndexOutOfBoundsException"><![CDATA[Given user navigates to home page...........................................passed
When I enter product name as "book".........................................failed
And user clicks on search button............................................skipped

StackTrace:
java.lang.IndexOutOfBoundsException: Index: 1, Size: 1
	at java.util.ArrayList.rangeCheck(ArrayList.java:659)
	at java.util.ArrayList.get(ArrayList.java:435)
	at org.example.stepDefs.D04_searchStepDef.iEnterProductNameAs(D04_searchStepDef.java:32)
	at ✽.I enter product name as "book"(file:///C:/apache-maven-3.8.6/demonopcommerce/src/main/resources/features/F04_Search.feature:8)
]]></failure>
</testcase>
<testcase classname="Search Feature" name="user could search using product name 2" time="9.562">
<failure message="Index: 1, Size: 1" type="java.lang.IndexOutOfBoundsException"><![CDATA[Given user navigates to home page...........................................passed
When I enter product name as "laptop".......................................failed
And user clicks on search button............................................skipped

StackTrace:
java.lang.IndexOutOfBoundsException: Index: 1, Size: 1
	at java.util.ArrayList.rangeCheck(ArrayList.java:659)
	at java.util.ArrayList.get(ArrayList.java:435)
	at org.example.stepDefs.D04_searchStepDef.iEnterProductNameAs(D04_searchStepDef.java:32)
	at ✽.I enter product name as "laptop"(file:///C:/apache-maven-3.8.6/demonopcommerce/src/main/resources/features/F04_Search.feature:8)
]]></failure>
</testcase>
<testcase classname="Search Feature" name="user could search using product name 3" time="9.997">
<failure message="Index: 1, Size: 1" type="java.lang.IndexOutOfBoundsException"><![CDATA[Given user navigates to home page...........................................passed
When I enter product name as "nike".........................................failed
And user clicks on search button............................................skipped

StackTrace:
java.lang.IndexOutOfBoundsException: Index: 1, Size: 1
	at java.util.ArrayList.rangeCheck(ArrayList.java:659)
	at java.util.ArrayList.get(ArrayList.java:435)
	at org.example.stepDefs.D04_searchStepDef.iEnterProductNameAs(D04_searchStepDef.java:32)
	at ✽.I enter product name as "nike"(file:///C:/apache-maven-3.8.6/demonopcommerce/src/main/resources/features/F04_Search.feature:8)
]]></failure>
</testcase>
<testcase classname="Search Feature" name="user could search for product using sku" time="14.785">
<system-out><![CDATA[Given user navigates to home page...........................................passed
When I enter product name using sku as "SCI_FAITH"..........................passed
And user clicks on search button............................................passed
]]></system-out>
</testcase>
<testcase classname="Search Feature" name="user could search for product using sku 2" time="15.016">
<system-out><![CDATA[Given user navigates to home page...........................................passed
When I enter product name using sku as "APPLE_CAM"..........................passed
And user clicks on search button............................................passed
]]></system-out>
</testcase>
<testcase classname="Search Feature" name="user could search for product using sku 3" time="16.16">
<system-out><![CDATA[Given user navigates to home page...........................................passed
When I enter product name using sku as "SF_PRO_11"..........................passed
And user clicks on search button............................................passed
]]></system-out>
</testcase>
<testcase classname="hoverCategories Feature" name="use Actions class in Selenium framework to hover on one of main categories then click on sub category" time="9.432">
<failure message="invalid selector: Unable to locate an element with the xpath expression //ul[@class=\&quot;top-menu notmobile\&quot;]/li because of the following error:&#10;SyntaxError: Failed to execute 'evaluate' on 'Document': The string '//ul[@class=\&quot;top-menu notmobile\&quot;]/li' is not a valid XPath expression.&#10;  (Session info: chrome=103.0.5060.114)&#10;For documentation on this error, please visit: https://selenium.dev/exceptions/#invalid_selector_exception&#10;Build info: version: '4.3.0', revision: 'a4995e2c09*'&#10;System info: host: 'DESKTOP-9J129ML', ip: '192.168.1.12', os.name: 'Windows 10', os.arch: 'amd64', os.version: '10.0', java.version: '1.8.0_281'&#10;Driver info: org.openqa.selenium.chrome.ChromeDriver&#10;Command: [a2895ea203875a5caa320d09969229c3, findElements {using=xpath, value=//ul[@class=\&quot;top-menu notmobile\&quot;]/li}]&#10;Capabilities {acceptInsecureCerts: false, browserName: chrome, browserVersion: 103.0.5060.114, chrome: {chromedriverVersion: 103.0.5060.53 (a1711811edd7..., userDataDir: C:\Users\ENGAMR~1\AppData\L...}, goog:chromeOptions: {debuggerAddress: localhost:57060}, networkConnectionEnabled: false, pageLoadStrategy: normal, platformName: WINDOWS, proxy: Proxy(), se:cdp: ws://localhost:57060/devtoo..., se:cdpVersion: 103.0.5060.114, setWindowRect: true, strictFileInteractability: false, timeouts: {implicit: 0, pageLoad: 300000, script: 30000}, unhandledPromptBehavior: dismiss and notify, webauthn:extension:credBlob: true, webauthn:extension:largeBlob: true, webauthn:virtualAuthenticators: true}&#10;Session ID: a2895ea203875a5caa320d09969229c3" type="org.openqa.selenium.InvalidSelectorException"><![CDATA[When hover on one of main categories........................................failed
And click on sub category...................................................skipped

StackTrace:
org.openqa.selenium.InvalidSelectorException: invalid selector: Unable to locate an element with the xpath expression //ul[@class=\"top-menu notmobile\"]/li because of the following error:
SyntaxError: Failed to execute 'evaluate' on 'Document': The string '//ul[@class=\"top-menu notmobile\"]/li' is not a valid XPath expression.
  (Session info: chrome=103.0.5060.114)
For documentation on this error, please visit: https://selenium.dev/exceptions/#invalid_selector_exception
Build info: version: '4.3.0', revision: 'a4995e2c09*'
System info: host: 'DESKTOP-9J129ML', ip: '192.168.1.12', os.name: 'Windows 10', os.arch: 'amd64', os.version: '10.0', java.version: '1.8.0_281'
Driver info: org.openqa.selenium.chrome.ChromeDriver
Command: [a2895ea203875a5caa320d09969229c3, findElements {using=xpath, value=//ul[@class=\"top-menu notmobile\"]/li}]
Capabilities {acceptInsecureCerts: false, browserName: chrome, browserVersion: 103.0.5060.114, chrome: {chromedriverVersion: 103.0.5060.53 (a1711811edd7..., userDataDir: C:\Users\ENGAMR~1\AppData\L...}, goog:chromeOptions: {debuggerAddress: localhost:57060}, networkConnectionEnabled: false, pageLoadStrategy: normal, platformName: WINDOWS, proxy: Proxy(), se:cdp: ws://localhost:57060/devtoo..., se:cdpVersion: 103.0.5060.114, setWindowRect: true, strictFileInteractability: false, timeouts: {implicit: 0, pageLoad: 300000, script: 30000}, unhandledPromptBehavior: dismiss and notify, webauthn:extension:credBlob: true, webauthn:extension:largeBlob: true, webauthn:virtualAuthenticators: true}
Session ID: a2895ea203875a5caa320d09969229c3
	at sun.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method)
	at sun.reflect.NativeConstructorAccessorImpl.newInstance(NativeConstructorAccessorImpl.java:62)
	at sun.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:45)
	at java.lang.reflect.Constructor.newInstance(Constructor.java:423)
	at org.openqa.selenium.remote.codec.w3c.W3CHttpResponseCodec.createException(W3CHttpResponseCodec.java:200)
	at org.openqa.selenium.remote.codec.w3c.W3CHttpResponseCodec.decode(W3CHttpResponseCodec.java:133)
	at org.openqa.selenium.remote.codec.w3c.W3CHttpResponseCodec.decode(W3CHttpResponseCodec.java:53)
	at org.openqa.selenium.remote.HttpCommandExecutor.execute(HttpCommandExecutor.java:184)
	at org.openqa.selenium.remote.service.DriverCommandExecutor.invokeExecute(DriverCommandExecutor.java:167)
	at org.openqa.selenium.remote.service.DriverCommandExecutor.execute(DriverCommandExecutor.java:142)
	at org.openqa.selenium.remote.RemoteWebDriver.execute(RemoteWebDriver.java:569)
	at org.openqa.selenium.remote.ElementLocation$ElementFinder$2.findElements(ElementLocation.java:179)
	at org.openqa.selenium.remote.ElementLocation.findElements(ElementLocation.java:102)
	at org.openqa.selenium.remote.RemoteWebDriver.findElements(RemoteWebDriver.java:402)
	at org.openqa.selenium.remote.RemoteWebDriver.findElements(RemoteWebDriver.java:394)
	at org.example.pages.P03_homePage.hoverCategories(P03_homePage.java:23)
	at org.example.stepDefs.D05_hoverCategoriesStepDef.hoverOnOneOfMainCategories(D05_hoverCategoriesStepDef.java:19)
	at ✽.hover on one of main categories(file:///C:/apache-maven-3.8.6/demonopcommerce/src/main/resources/features/F05_hoverCategories.feature:4)
]]></failure>
</testcase>
</testsuite>
